students/johnpharmd/lesson06/mailroomtest.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+import unittest
+import mailroom4
+import os
+import datetime
+
+
+class MailroomTest(unittest.TestCase):
+    def make_test_list(self):
+        self.test_list = mailroom4.get_donor_list()
+        return self.test_list
+
+    def test_get_donor_list(self):
+        self.make_test_list()
+        assert type(self.test_list) == list
+
+    def test_donor_list_strings(self):
+        self.make_test_list()
+        for s in self.test_list:
+            if '-' not in s:
+                assert s.isalpha()
+            else:
+                assert s.replace('-', '').isalpha()
+
+    def test_prompt_title(self):
+        self.test_title = mailroom4.prompt_title()
+        assert self.test_title in ('Prof.', 'Dr.', 'Ms.', 'Mr.')
```

Comment:
unit tests are typically run in automated way, there is usually a system like Jenkins for example that will run your tests on each commit or at build time. Because of that your unit tests cannot typically rely on user interaction.

so how does something like this get tested? that's where we get into the mocking and patching, it is a bit of an advanced concept but simply put you would be "mocking" out (faking) the user input portion and would supply predefined mocked input values for your code.